### ISSUE TITLE
AzureIngressProhibitedTarget not working as expected - AG resources still removed by AKS

### ISSUE THREAD

**Issue by marc-mueller** [Azure/application-gateway-kubernetes-ingress#940]
**Describe the bug**
In our deployment, we have a single service which is run by VMSS and needs to be available through the same application gateway as AKS is using. We added a AzureIngressProhibitedTarget CRD with the domain name of the VMSS service. But after every AKS deployment, the VMSS objects are completely removed by AKS.

**To Reproduce**
- Crate a AzureIngressProhibitedTarget in default namespace with domain hostname <abc>
- Create the following AG objects by az cli: ssl-cert, address-pool, frontend-port, http-listener, probe, https-settings, rule --> the http-listener is pointing to the hostname <abc>
- Do an AKS deployment, the above manually added AG objects are removed

Did I miss something? Do I need to pay special attention to some details while manually creating the AG objects? The only thing mentioned in the documentation is the hostname which has to match. 

Thanks for your help!

**Comment by 3quanfeng** [comment]
Hi @marc-mueller , May I ask which version of AGIC you deployed? there is a related bug fixed in v1.2.0-rc3

**Comment by marc-mueller** [comment]
Hi @3quanfeng, we are currently using 1.0.0.

**Comment by 3quanfeng** [comment]
Hi @marc-mueller , did you have prohibit-all-targets deployed as well before your own prohibited policy?
also, could you please kindly share your sanitized prohibited policy?

**Comment by marc-mueller** [comment]
No, I just deployed my custom prohibited policy. AKS was setup before and then later on we needed the manually managed endpoints in AG. If I understood the documentation correctly, the custom policy is the only thing needed in this case, right?

We used the following policy:
```
apiVersion: "appgw.ingress.k8s.io/v1"
kind: AzureIngressProhibitedTarget
metadata:
  name: manually-configured-xy-service
spec:
  hostname: xy.mydomain.com
```

**Comment by 3quanfeng** [comment]
Thanks much @marc-mueller .
Could you please help check your prohibited policies by "kubectl get AzureIngressProhibitedTarget". when you upgrade AGIC with shared enabled(assuming you upgrade AGIC with shared enabled via helm), a default prohibit-all-targets policy will be installed, you can then install your own policy as you mentioned above, but to make your policy work, you will need to delete the default one which is prohibit-all-targets.
Please let us know if you still run into the same issue by following the steps above(or the doc [here](https://github.com/Azure/application-gateway-kubernetes-ingress/blob/master/docs/setup/install-existing.md#broaden-permissions)). Thanks!

**Comment by akshaysngupta** [comment]
@marc-mueller Can you please check if the prohibited target is being picked by AGIC by checking the logs ?
Also, can you upgrade the installation to 1.2.0 ?

**Comment by vishal8k** [comment]
@marc-mueller Which namespace is your AGIC deployed? Is AGIC and prohibited targets in the same namespace? Was AGIC configured to look at all namespaces during installation? Need to clear these out first before going deeper.

**Comment by williamalvvc** [comment]
Hey guys, i have the exact same problem.

According to the docs, if there are a previously deployed AGIC, using a custom prohibited policy, should be enough.

But as soon as I made a k8s deployment the manually created AG objects are deleted. Do you have any ideas? Could you solve the issue?

**Comment by adamcarter81** [comment]
Hi all - I have just run into this as well - I have AGIC deployed (1.5.0), but now needed to add a custom hostname not handled by AKS. Easy I thought, I remember AGIC could ignore certain hostnames, and the documentation states for an existing AGIC installation we just need to create a CRD for the hostname. I did in the default namespace (as per AGIC), but sure enough - on the next refresh it just deletes everything associated with it (listener, rules, backend etc).

Am I missing a step somewhere?

**Comment by adamcarter81** [comment]
I think I have solved this - I still needed to set the shared=true in the helm values - once I did this and deleted the prohibitall CRD it kept my custom domain. The documentation should be updated for the section Enable for an existing AGIC installation - that still the shared setting needs to be updated

**Comment by manikanta1994** [comment]
Hi Team,

I had velero in my aks cluster, When the AKS cluster is at 1.21.9 version everything worked fine, We had upgraded to  1.22.4 -->1.23.5. Present cluster version is 1.23.5. Once after upgrade my backups are failing due to below error, Could some one please look into this and help me on this ?
<img width="1792" alt="image" src="https://user-images.githubusercontent.com/33510091/171794310-531eb6c9-2387-4d31-b3b5-e3ba0dda4ae2.png">

**Comment by shansted** [comment]
I went down this path today, and tracked this down to a configmap setting the helm chart must set.

https://github.com/Azure/application-gateway-kubernetes-ingress/blob/master/helm/ingress-azure/templates/configmap.yaml

APPGW_ENABLE_SHARED_APPGW: 'true'

After setting this, check for the following entries in the AGIC pod logs.

[brownfield] Existing Blacklisted Rules AGIC will retain:
[brownfield] Existing Rules AGIC will remove: 

These were not present before setting the flag.

**Comment by bjburke002** [comment]
> I went down this path today, and tracked this down to a configmap setting the helm chart sets.
> 
> https://github.com/Azure/application-gateway-kubernetes-ingress/blob/master/helm/ingress-azure/templates/configmap.yaml
> 
> APPGW_ENABLE_SHARED_APPGW: 'true'
> 
> After setting this, check for the following entries in the AGIC pod logs.
> 
> [brownfield] Existing Blacklisted Rules AGIC will retain: [brownfield] Existing Rules AGIC will remove:
> 
> These were not present before setting the flag.

Thanks for this--couldn't find it for the life of me but this cleared up an issue I was dealing with.

**Comment by pienczyk** [comment]
Did anyone tried to integrate the AGIC with an existing, shared Application Gateway via the Terraform?

I tried this:

AKS deployment with option to enabled AGIC [TF doc](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/kubernetes_cluster#ingress_application_gateway):

```
resource "azurerm_kubernetes_cluster" "main" {
..
ingress_application_gateway {
        gateway_id = "The ID of the existing Application Gateway"
}

...
}
```

Then I have deployed the [AzureIngressProhibitedTarget CRD](https://github.com/Azure/application-gateway-kubernetes-ingress/blob/master/crds/AzureIngressProhibitedTarget-v1-CRD-v1.yaml) and a test AzureIngressProhibitedTarget for:
apiVersion: "appgw.ingress.k8s.io/v1"
```
kind: AzureIngressProhibitedTarget
metadata:
  name: testhost.dev.example.com
spec:
  hostname: testhost.dev.example.com
```

All seems to be deployed ok but the AGIC still overwrites the Application Gateway configuration created for the "testhost.dev.example.com".

I tried to modify the "ingress-appgw-cm" configMap and add the APPGW_ENABLE_SHARED_APPGW set to "true" but it is overwritten after some time. The configMap has the label:  "addonmanager.kubernetes.io/mode: Reconcile" I tried to overwrite the label value but no luck.  

Versions:
AKS Kuberntets version - 1.30.0
hashicorp/azurerm - 3.112.0
AGIC - 1.7.5
